### PR TITLE
Try to speed up our CI a little

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

Seeing CI pass is enough testing for this change

## Trade-offs

- Could have used 3. Party `npm ci` alternative, but the speedup is similar + we would get a new dependency
- Thought about trying to run everything on a different host as npm install is now half the time, but that would be a bigger change I would rather do at a later point in time.

### Get node modules
```
npm:        69.00 real        96.28 user        69.38 sys
npm ci:     36.47 real        28.50 user        19.65 sys
```

### Enabled parallelization

Before: 1h 10min in best case
After: 40min in best case

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -31,13 +31,19 @@ pipeline {
 
     stage("Build") {
       steps {
-        sh "npm --unsafe-perm install"
-        sh "npm run build"
```
Comment (nLight) [inline Jenkinsfile]: We still need to keep linting on the CI + switching to `build-assets` will omit `prebuild` stage.

```
"build-assets": "npm run clean && lingui compile && NODE_ENV=production webpack --colors --config ./webpack/webpack.production.js",
./dist/index.html {} \\;",
"prebuild": "./scripts/validate-tests",
"build": "npm run lint && ([ -z \"${npm_config_externalplugins}\" ] || npm run lint-plugins) && npm test && npm run build-assets && npm run validate-build",
```

Comment (DanielMSchmidt) [reply Jenkinsfile]: I see, but if I'd move prebuild to pretest this should be fine, shouldn't it?

Comment (juliangieseke) [reply Jenkinsfile]: Yup, thats better anyway

**Comment by DanielMSchmidt** [comment]
@juliangieseke @nLight I think I got your feedback now

File: Jenkinsfile (hunk 1 above)
Comment (nLight) [reply Jenkinsfile]: I don't see the comment being resolved

Comment (DanielMSchmidt) [reply Jenkinsfile]: See the package.json

Comment (nLight) [reply Jenkinsfile]: We still skipping  `validate-build` step. That could mean once we merge master is broken. 
Shall we move it to `postbuild-assets?`

**Comment by nLight** [comment]
```

[cos-ui-oss-pipeline_PR-3394-YYHNNIFCFYR6D4RABSJKSQVXLXH3C26TBIRECMDDC3DJCTUO6MYA] Running shell script

+ npm --unsafe-perm ci

npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.

npm ERR! 

npm ERR! 

npm ERR! Invalid: lock file's @dcos/ui-kit@1.16.1 does not satisfy @dcos/ui-kit@1.17.1

npm ERR! 



npm ERR! A complete log of this run can be found in:

npm ERR!     /root/.npm/_logs/2018-11-23T12_56_19_594Z-debug.log

script returned exit code 1
```

**Comment by DanielMSchmidt** [comment]
This was an old try to speed up CI that was not really helpful

**Comment by nLight** [comment]
I believe using `npm ci` _is_ helpful, the surrounding stuff is out of scope :)

**Comment by DanielMSchmidt** [comment]
It only helps with like 3 minutes and it was done without Jira issue, so let's try to do it properly instead 👍

**Pull request closed without merging**